ag_data/simulator.py

```diff
@@ -1,146 +0,0 @@
-)
-from ag_data import utilities
-
-
-class Simulator:
-
-    event = None
-    sensor = None
-    engine = MeasurementIngestionEngine()
-
-    def setUp(self, event, sensor):
-        self.event = event
-        self.sensor = sensor
-
-    def logProcessedSingleMeasurement(self, timestamp, value):
-        utilities.assertEvent(self.event)
-        utilities.assertSensor(self.sensor)
-
-        measurement_data = MeasurementExchange(
-            event=self.event, timestamp=timestamp, sensor=self.sensor, reading=value
-        )
-
-        return self.engine.saveMeasurement(measurement_data)
-
-    def logSingleMeasurement(self, timestamp):
-
-        utilities.assertEvent(self.event)
-        utilities.assertSensor(self.sensor)
-
-        reading = self.generateMeasurementReading()
-
-        return self.logProcessedSingleMeasurement(timestamp, reading)
-
-    def checkSensorType(self, typeID):
-        return self.sensor.type_id.id == typeID
-
-    def generateMeasurementReading(self):
-        utilities.assertSensor(self.sensor)
-
-        reading = {}
-
-        if self.checkSensorType(0):
-            reading = {"side": (random() < 0.5)}
-
-        elif self.checkSensorType(2):
-            reading = {"reading": gauss(23, 1)}
-
-        elif self.checkSensorType(4):
-            reading = {"internal": gauss(15, 3), "external": gauss(20, 2)}
-
-        elif self.checkSensorType(6):
-            reading = {"volumetricFlow": gauss(0.2, 0.15)}
-
-        elif self.checkSensorType(8):
-            reading = {"sample": gauss(0.5, 0.5)}
-
-        else:
-            raise ValueError(f"Unsupported sensor type ({self.sensor.type_id})")
-
-        return reading
-
-    def logMeasurementsInThePastSeconds(
-        self, seconds, frequencyInHz, printProgress=True
-    ):
-        """Create as many measurements from the past till current time as needed
-        with the specified time range and frequency.
-        """
-
-        startTime = timezone.now()
-        earliestReading = startTime - timezone.timedelta(seconds=seconds)
-        sampleInterval = 1 / frequencyInHz
-        count = 0
-        totalMeasurements = int(seconds * frequencyInHz)
-
-        for count in range(totalMeasurements):
-            timeAtReading = earliestReading + timezone.timedelta(
-                seconds=count / frequencyInHz + gauss(0, sampleInterval)
-            )
-            self.logSingleMeasurement(timeAtReading)
-            if printProgress is True and count % 1000 == 0:
-                print(
-                    f"({(count / totalMeasurements * 100):3.3f}%"
-                    f") Created {str(count)} measurements"
-                )
-
-        if printProgress is True:
-            print(f"(100% done!) Created {totalMeasurements} measurements")
-
-        endTime = timezone.now()
-        if printProgress is True:
-            print("Time elapsed: " + str(endTime - startTime))
-
-    def logLiveMeasurements(self, frequencyInHz, sleepTimerInSeconds=0):
```

Comment:
I mean, is there any similar functionality in the new design that creates live measurements, which I believe contributes a great value to the development process. I am open to any ideas that improves the development workflow.

Reply:
1. You can do TDD. Write tests for expected behavior. Run the test.
2. And after you're done, you can test overall functionality with the simulator script (end to end).

Reply:
We can do that. But until we can do that with the simulator script, I think we should keep it for now.

Reply:
I agree. You're right in a way. We can use the simulator script to test only after we integrate formula things to incoming data flow.

At the same time, I don't think integrating formula to the api won't take long. Rather, my concern is that the complexity from keeping simulator.py and associated files might slow down the development progress. Because it's harder to understand than what it supposed to be.

My idea is not to keep those files and to make all these things simple so that we can quickly integrate formula to the project, which is not only easier to develop but also easier to be reviewed by others.

Reply:
Sure. The priority is to integrate the formula pipeline into the API. We can have radio port simulator updated afterwards.